### DISCUSSION TITLE
uac2_headset example with 2 input channels

### DISCUSSION THREAD

**Discussion by Lurcy38** [hathach/tinyusb#2312] [General]
Hello, 
Before I am completely formed to the art to build usb descriptors,  I would like to modify the uac2_headset example . The modification consists to create a input with 2 channels rather than 1. I saw in the example tusb_config.h a define 
#define CFG_TUD_AUDIO_FUNC_1_N_CHANNELS_TX                           1
So, I modified by value hopping that the input will be changed from 1 to 2 channels 
But it is not. The usb device is non longer recognized 
Any tips to fix this issue ? 
Thanks & Regards

**Comment by Lurcy38** [comment]
it seams i fixed my issue using this modification
#define CFG_TUD_AUDIO_FUNC_1_N_CHANNELS_TX                           **2**
and in the TUD_AUDIO_DESC_INPUT_TERM _nchannelslogical = CFG_TUD_AUDIO_FUNC_1_N_CHANNELS_TX

    TUD_AUDIO_DESC_INPUT_TERM(/*_termid*/ UAC2_ENTITY_MIC_INPUT_TERMINAL, /*_termtype*/ AUDIO_TERM_TYPE_IN_GENERIC_MIC, /*_assocTerm*/ 0x00, /*_clkid*/ UAC2_ENTITY_CLOCK, /*_nchannelslogical 0x01*/ **CFG_TUD_AUDIO_FUNC_1_N_CHANNELS_TX**, /*_channelcfg*/ AUDIO_CHANNEL_CONFIG_NON_PREDEFINED, /*_idxchannelnames*/ 0x00, /*_ctrl*/ 0 * (AUDIO_CTRL_R << AUDIO_IN_TERM_CTRL_CONNECTOR_POS), /*_stridx*/ 0x00),\

**Comment by HiFiPhile** [comment]
There are other modifications needed in the descriptor, at least feature unit & `nchannelsphysical`.
You can compare mic 1ch & 4ch descriptors:

https://github.com/hathach/tinyusb/blob/f84eafcaa2138a360e0baf8271b28f45cf120961/src/device/usbd.h#L442
https://github.com/hathach/tinyusb/blob/f84eafcaa2138a360e0baf8271b28f45cf120961/src/device/usbd.h#L491

**Reply by Lurcy38** [reply]
many thanks, may application seems to work fine now. I still have a last issue. i need to detect if the speaker or the microphone is streaming ( or connected)
i didn't see any api to check this state. 
any tips ?

**Reply by HiFiPhile** [reply]
You can check if alt settings is not 0 in `tud_audio_set_itf_cb()`:
https://github.com/hathach/tinyusb/blob/a6b29ae57c3cb4408eb350d1a9f4e03897726193/examples/device/uac2_headset/src/main.c#L344
